arangod/Cluster/RebootTracker.h

```diff
@@ -23,119 +23,83 @@
 
 #pragma once
 
-#include "Basics/Mutex.h"
 #include "Cluster/CallbackGuard.h"
 #include "Cluster/ClusterTypes.h"
 #include "Containers/FlatHashMap.h"
 
 #include <map>
 #include <memory>
 #include <type_traits>
-#include <unordered_map>
 #include <vector>
+#include <mutex>
 
 namespace arangodb {
 
 class SupervisedScheduler;
 
 namespace velocypack {
+
 class Builder;
 class Slice;
-}  // namespace velocypack
 
+}  // namespace velocypack
 namespace cluster {
 
 // Note:
 // Instances of this class must be destructed during shutdown before the
 // scheduler is destroyed.
 class RebootTracker {
  public:
-  using Callback = std::function<void(void)>;
   using SchedulerPointer = SupervisedScheduler*;
-
-  class PeerState {
-   public:
-    PeerState(ServerID serverId, RebootId rebootId)
-        : _serverId{std::move(serverId)}, _rebootId{rebootId} {}
```

Comment:
> Please can you show example where we use this struct and that give more readable code

It's about semantics, not convenience of use. To explain what I mean by that:

This is about the "user"-facing API, i.e. `callMeOnChange`. This binds a callback to a `PeerState`, and if that state changes, the callback is executed. In a perfect world, this would be an opaque value from the beginning: e.g., a REST request would allow you to read the `PeerState` of the client (think for example of an Coordinator, setting up AQL snippets on DBServers). You can hand this over to the RebootTracker, but do nothing else with it - possibly not even (easily) construct a `PeerState`. That would prevent pitfalls, like someone reading the RebootId of the client from the AgencyCache instead of getting it from the client itself.

Now nothing of that is currently there, but the least we can do is emphasize that these values belong together, and form the smallest unit that makes sense: Neither the ServerID has any meaning on its own, nor the RebootID.